File: sdk/java/src/main/java/com/astrovox/sdk/AstrovoxClient.java
package com.astrovox.sdk;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.List;
import java.util.Map;

public class AstrovoxClient {
    private final String apiKey;
    private final String baseUrl;
    private final HttpClient httpClient;

    public AstrovoxClient(String apiKey, String baseUrl) {
        this.apiKey = apiKey;
        this.baseUrl = baseUrl != null ? baseUrl : "https://api.astrovox.ai/v1";
        this.httpClient = HttpClient.newHttpClient();
    }

    private HttpRequest.Builder requestBuilder(String method, String path) {
        return HttpRequest.newBuilder()
                .uri(URI.create(baseUrl + path))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .timeout(Duration.ofSeconds(30));
    }

    public String sendMessage(String conversationId, String message, String model) throws IOException, InterruptedException {
        String body = String.format("{\"conversation_id\":\"%s\",\"message\":\"%s\",\"model\":\"%s\"}", conversationId, message, model);
        HttpRequest request = requestBuilder("POST", "/chat/message").POST(HttpRequest.BodyPublishers.ofString(body)).build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public String createConversation(String title, String model) throws IOException, InterruptedException {
        String body = String.format("{\"title\":\"%s\",\"model\":\"%s\"}", title, model);
        HttpRequest request = requestBuilder("POST", "/conversations").POST(HttpRequest.BodyPublishers.ofString(body)).build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public String listConversations() throws IOException, InterruptedException {
        HttpRequest request = requestBuilder("GET", "/conversations").build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public String healthCheck() throws IOException, InterruptedException {
        HttpRequest request = requestBuilder("GET", "/health").build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public String getApiKey() {
        return apiKey;
    }

    public String getBaseUrl() {
        return baseUrl;
    }
}

File: sdk/templates/csharp/AstrovoxClient.cs
"""C# SDK client template."""
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace AstrovoxSDK;

public class Client
{
    public string ApiKey { get; }
    public string BaseUrl { get; }
    private readonly HttpClient _http;

    public Client(string apiKey, string baseUrl = "https://api.astrovox.ai/v1")
    {
        ApiKey = apiKey;
        BaseUrl = baseUrl;
        _http = new HttpClient();
        _http.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiKey}");
        _http.DefaultRequestHeaders.Add("Accept", "application/json");
    }

    private async Task<HttpResponseMessage> SendAsync(HttpMethod method, string path, object? body = null)
    {
        var request = new HttpRequestMessage(method, BaseUrl + path);
        if (body != null)
        {
            request.Content = new StringContent(JsonSerializer.Serialize(body), Encoding.UTF8, "application/json");
        }
        return await _http.SendAsync(request);
    }

    public async Task<string> SendMessageAsync(string conversationId, string message, string model = "gpt-4")
    {
        var resp = await SendAsync(HttpMethod.Post, "/chat/message", new { conversation_id = conversationId, message, model });
        return await resp.Content.ReadAsStringAsync();
    }

    public async Task<string> CreateConversationAsync(string title, string model = "gpt-4")
    {
        var resp = await SendAsync(HttpMethod.Post, "/conversations", new { title, model });
        return await resp.Content.ReadAsStringAsync();
    }

    public async Task<string> ListConversationsAsync()
    {
        var resp = await SendAsync(HttpMethod.Get, "/conversations");
        return await resp.Content.ReadAsStringAsync();
    }

    public async Task<string> HealthCheckAsync()
    {
        var resp = await SendAsync(HttpMethod.Get, "/health");
        return await resp.Content.ReadAsStringAsync();
    }
}
